apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js

```diff
@@ -0,0 +1,555 @@
+import _ from 'lodash';
+
+const INIT = 'activitiesEditor/INIT';
+const ADD_ACTIVITY = 'activitiesEditor/ADD_ACTIVITY';
+const MOVE_ACTIVITY = 'activitiesEditor/MOVE_ACTIVITY';
+const SET_ACTIVITY = 'activitiesEditor/SET_ACTIVITY';
+const REMOVE_ACTIVITY = 'activitiesEditor/REMOVE_ACTIVITY';
+const UPDATE_ACTIVITY_FIELD = 'activitiesEditor/UPDATE_ACTIVITY_FIELD';
+const ADD_ACTIVITY_SECTION = 'activitiesEditor/ADD_ACTIVITY_SECTION';
+const MOVE_ACTIVITY_SECTION = 'activitiesEditor/MOVE_ACTIVITY_SECTION';
+const REMOVE_ACTIVITY_SECTION = 'activitiesEditor/REMOVE_ACTIVITY_SECTION';
+const UPDATE_ACTIVITY_SECTION_FIELD =
+  'activitiesEditor/UPDATE_ACTIVITY_SECTION_FIELD';
+const ADD_TIP = 'activitiesEditor/ADD_TIP';
+const ADD_LEVEL = 'activitiesEditor/ADD_LEVEL';
+const REMOVE_LEVEL = 'activitiesEditor/REMOVE_LEVEL';
+const CHOOSE_LEVEL = 'activitiesEditor/CHOOSE_LEVEL';
+const REORDER_LEVEL = 'activitiesEditor/REORDER_LEVEL';
+const MOVE_LEVEL_TO_ACTIVITY_SECTION =
+  'activitiesEditor/MOVE_LEVEL_TO_ACTIVITY_SECTION';
+const ADD_VARIANT = 'activitiesEditor/ADD_VARIANT';
+const SET_ACTIVE_VARIANT = 'activitiesEditor/SET_ACTIVE_VARIANT';
+const REMOVE_VARIANT = 'activitiesEditor/REMOVE_VARIANT';
+const SET_FIELD = 'activitiesEditor/SET_FIELD';
+const TOGGLE_EXPAND = 'activitiesEditor/TOGGLE_EXPAND';
+
+// NOTE: Position for Activities, Activity Sections and Levels is 1 based.
+
+export const init = (activities, levelKeyList) => ({
+  type: INIT,
+  activities,
+  levelKeyList
+});
+
+export const addActivity = (activityPosition, activityKey) => ({
+  type: ADD_ACTIVITY,
+  activityPosition,
+  activityKey
+});
+
+export const updateActivityField = (
+  activityPosition,
+  fieldName,
+  fieldValue
+) => ({
+  type: UPDATE_ACTIVITY_FIELD,
+  activityPosition,
+  fieldName,
+  fieldValue
+});
+
+export const updateActivitySectionField = (
+  activityPosition,
```

Comment:
Is there a reason to use the position values to identify an activity / section in these actions rather than key? In general I feel like using keys would be more debuggable / readable.

Reply:
I followed the way that things were set up for the script edit gui. I think that key might be better but since I have things completely working (see follow up https://github.com/code-dot-org/code-dot-org/pull/36627) I don't know that its worth the work to change it right now.